WordPress/Classes/Services/Store.swift

```diff
@@ -22,53 +22,147 @@ class StoreKitTransactionObserver: NSObject, SKPaymentTransactionObserver {
 // stored property since it's a generic class.
 struct StoreKitCoordinator {
     static let instance = StoreCoordinator(store: StoreKitStore())
+    
+    static let TransactionDidFinishNotification = "StoreCoordinatorTransactionDidFinishNotification"
+    static let TransactionDidFailNotification   = "StoreCoordinatorTransactionDidFailNotification"
+    static let NotificationProductIdentifierKey = "StoreCoordinatorNotificationProductIdentifierKey"
+    static let NotificationErrorDescriptionKey  = "StoreCoordinatorNotificationErrorDescriptionKey"
 }
 
+typealias PendingPayment = (planID: PlanID, productID: String, siteID: Int)
+
 class StoreCoordinator<S: Store> {
     let store: S
-    var pendingPayment: (plan: Plan, siteID: Int)? = nil
+    
+    var pendingPayment: PendingPayment? {
+        set {
+            let defaults = NSUserDefaults.standardUserDefaults()
+            
+            if let pending = newValue {
+                defaults.setInteger(pending.planID, forKey: UserDefaultsKeys.pendingPaymentPlanID)
+                defaults.setObject(pending.productID, forKey: UserDefaultsKeys.pendingPaymentProductID)
+                defaults.setInteger(pending.siteID, forKey:UserDefaultsKeys.pendingPaymentSiteID)
+            } else {
+                defaults.removeObjectForKey(UserDefaultsKeys.pendingPaymentPlanID)
+                defaults.removeObjectForKey(UserDefaultsKeys.pendingPaymentProductID)
+                defaults.removeObjectForKey(UserDefaultsKeys.pendingPaymentSiteID)
+            }
+        }
+        
+        get {
+            let defaults = NSUserDefaults.standardUserDefaults()
+            let planID = defaults.integerForKey(UserDefaultsKeys.pendingPaymentPlanID)
+            let productID = defaults.stringForKey(UserDefaultsKeys.pendingPaymentProductID)
+            let siteID = defaults.integerForKey(UserDefaultsKeys.pendingPaymentSiteID)
+            
+            guard let product = productID where planID != 0 && siteID != 0 else { return nil }
+            
+            return (planID, product, siteID)
+        }
+    }
 
     init(store: S) {
         self.store = store
     }
 
+    /// The store coordinator will post a notification on a successful or failed purchase:
+    /// - `StoreKitCoordinator.TransactionDidFinishNotification` on success. 
+    ///   The notification's `userInfo` will contain the productID of the purchased 
+    ///   product under `StoreKitCoordinator.NotificationProductIdentifierKey`.
+    /// - `StoreKitCoordinator.TransactionDidFailNotification` on failure.
+    ///   The notification's `userInfo` will also contain the productID of the attempted
+    ///   purchased product, as well as a localized error message under `StoreKitCoordinator.NotificationErrorDescriptionKey`.
```

Comment:
Since you're adding documentation about the posted notifications, how about documenting the rest of the method. Although I'm thinking that the notification info might belong in the class documentation instead.